### DISCUSSION TITLE
Nvidia Jetson Nano	 - Submissions for EXPORT competition

### DISCUSSION THREAD

**Discussion by batrlatom** [ultralytics/yolov5#3426] [Show and tell]
## <div align="center">Compete and Win</div>

Use this discussion thread for YOLOv5 🚀 EXPORT Competition submissions in the **Nvidia Jetson Nano** category. Good luck!

<p align="center">
  <a href="https://github.com/ultralytics/yolov5/discussions/3213">
  <img width="850" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/banner-export-competition.png"></a>
</p>

**Comment by juantoca** [comment]
Hi. We are preparing a submission for the competition. Even though it currently works as expected and it will be finished soon, we would like to know which Jetpack version  are we expected to support. Are we supposed to support the last stable release(4.5.\*) or shall we support also the older versions of Jetpack(4.4.\*)?

**Reply by batrlatom** [reply]
4.5.* is fine to us

**Comment by sean-wade** [comment]
Where can I see the code submitted by the contributers? Must we wait until the end of the competition to see it?

**Reply by batrlatom** [reply]
Participants should place their submissions here. There are no entries to this category yet, therefore, no links are visible yet.

**Reply by sean-wade** [reply]
Ok, thanks a lot !!!

**Comment by sainikhileshreddy** [comment]
Is Jetson Nano 2GB Developer Kit eligible for competition?

**Reply by batrlatom** [reply]
Yes, we would love to see YOLOv5 on 2GB kit!

**Comment by deepconsc** [comment]
Coded & wrote tutorial for Jetson Nano 🙌

Needs a little review and refactor, which you'll be able to see in the end of July 12-16 week.

✅ Bonus l - Includes INT8 conversion & calibration tutorial.
✅ Bonus ll - as I've personally integrated distributed inferences with Xavier mini-nodes in couple of organizations, I guess some of our colleagues would find it helpful to have clear walkthrough and documentation on Jetson Xavier. Plus, INT8 acceleration on it is INSANE! I'll add its documentation as well.

https://github.com/deepconsc/tensorrt_yolov5

**Reply by batrlatom** [reply]
Thanks for your entry, we will test it as soon as possible. We changed the date of competition ending, so you can still work on the codebase.

**Reply by deepconsc** [reply]
Glad to hear! As I read evaluation is planned between 1-16 September. I'll update the repository to the final version much earlier than that 🙌🏻

**Reply by shivamswarnkar** [reply]
@deepconsc did you benchmark your solution on Nanor, Xavier, or any other devices? It'd be great if you could share that (I'm trying to optimize overall inference pipeline)

**Comment by MAli-Farooq** [comment]
Hi 

Can I take part in this competition using Nvidia Xavier development kit ? I have implemented it and it's working good .

**Reply by batrlatom** [reply]
Hi, xavier and nano has almost the same software stack. It is fine to develop on xavier, it will work quite well. But we will test it on jetson nano, since it is much more affordable piece of hardware. We are looking fornyour submission

**Comment by shivamswarnkar** [comment]
Hi,

@aditya-dl  and I have worked on this submission for Nvidia Jetson Nano. 

https://github.com/BlueMirrors/Yolov5-TensorRT

We are getting identical outputs as PyTorch Inference(FP16). This uses letterboxing (so no need to forcefully resize) and you can directly test it on [Colab](https://colab.research.google.com/drive/1tXLk2KFZkXQ7SpTBbmQ_Y43Eo1_34Rsf?usp=sharing) as well (we got up to 165FPS on Colab T4 GPU in benchmarking).

P.S. We are working on int8 and batching support.

**Reply by batrlatom** [reply]
We are going to start the evaluation process next week. Stay tuned!

**Comment by alxmamaev** [comment]
I made experiments with yolov5 on mu Jetson Nano, and made docker container to build tensorrt models. Also I made python wrapper with NMS for it. Wrapper allows you process photos, batch of photos, videos and camera streaming. Demos included.

https://github.com/alxmamaev/jetson_yolov5_tensorrt

**Reply by batrlatom** [reply]
thank you for your submission. We are going to test it next week

**Reply by alxmamaev** [reply]
Update: I'm update NMS algorithm for more comparability  with original implementation. Also I pushed docker image to docker hub, now you does not need build all by your own.

**Reply by stefani-kovachevska** [reply]
Hey @alxmamaev, I've tried to reach you at [hi@alxmamaev.me](mailto:hi@alxmamaev.me). Could you get in touch by replying or emailing me at [stefani@ultralytics.com](mailto:stefani@ultralytics.com)? Thanks!

**Reply by alxmamaev** [reply]
Hey, yes!

**Comment by jario-jin** [comment]
Submission for YOLOv5 EXPORT Competition (Jetson platform)

I tested the detection accuracy and speed on COCO dataset on nano (472 gflops), TX2 (1.33 tflops), Xavier NX (21 tops) and AgX Xavier (32 tops). Two different languages (Python and C++) are provided, and it is found that C++ is much more efficient than python. Finally, a detailed tutorial is provided.

https://github.com/jario-jin/yolov5-on-nvidia-jetson

**Reply by batrlatom** [reply]
Good job, looks promissing. We are going to start the evaluation next week!

**Comment by deepsworld** [comment]
`YOLOv5 EXPORT Competition (Jetson platform)`

- All the dependencies including ones to train and validate in a compact `docker container`.
- On the fly TensorRT engine build with `FP32, FP16, INT8 with calibration`.
- `No need` for any custom plugins.
- At least `4X inference speed up`...could be more with different devices and configs.
- `Dynamic shape input` is also supported

https://github.com/deepsworld/Yolo5Optimized

**Reply by glenn-jocher** [reply]
@deepsworld thank you for your submission!

**Comment by stefani-kovachevska** [comment]
Hi participants, 

Thank you all for your submissions, we have already reviewed them and are about to announce the winners! Please **send me an email** at [stefani@ultralytics.com](mailto:stefani@ultralytics.com) including your full name and GitHub username, so I can share more information with you. 

Stay tuned - the results are coming soon!

Stefani Kovachevska
_Project Manager at  Ultralytics_

**Comment by stefani-kovachevska** [comment]
Congratulations to @alxmamaev, our winner in the Nvidia Jetson Nano category! 🎉 
Check out the rest of the winners in the official announcement: https://ultralytics.com/article/export-competition-winners

Thank you all for participating and for making our community great! 

Stay amazing and keep creating! 🚀
_Team Ultralytics_
